Sources/KituraNet/ClientRequest.swift

```diff
@@ -535,11 +535,12 @@ public class ClientRequest {
      */
     public func end(close: Bool = false) {
         closeConnection = close
-
+        let percentEncodedURL = URL(string: self.percentEncodedURL)!
```

Comment:
Are you sure about the force unwrapping? It looks dangerously un-recoverable.

Reply:
I'm also a bit leery about it, but I was opting for minimal changes.  I think it will take quite a bit more restructuring to better handle these situations.